lisa/runners/legacy_runner.py

```diff
@@ -459,11 +455,25 @@ def discover_running_cases(self) -> List[Dict[str, str]]:
             case_match = self.CASE_RUNNING.match(line)
             if case_match:
                 name = case_match["name"]
-                current_case: Dict[str, str] = case_match.groupdict()
+                current_case: Dict[str, str] = {
+                    key: value
+                    for key, value in case_match.groupdict().items()
+                    if value is not None and value != ""
+                }
             image_match = self.CASE_IMAGE_LOCATION.match(line)
             if image_match:
                 location = image_match["location"]
-                current_case.update(image_match.groupdict())
+                current_case.update(
+                    {
+                        key: value
+                        for key, value in image_match.groupdict().items()
+                        if value is not None and value != ""
+                    }
+                )
+                # image for ARMImage, image2 for OsVHD
```

Comment:
good to see such comments!
BTW, we may consider rename "image" to "gallery_image", and "image2" to "vhd_image". that would be more clear.